src/main/java/uk/nhs/tis/sync/job/PostFundingSyncJob.java

```diff
@@ -0,0 +1,208 @@
+package uk.nhs.tis.sync.job;
+
+import com.google.common.base.Stopwatch;
+import com.google.common.collect.Sets;
+import com.transformuk.hee.tis.tcs.api.enumeration.Status;
+import com.transformuk.hee.tis.tcs.service.model.Post;
+import com.transformuk.hee.tis.tcs.service.model.PostFunding;
+import java.math.BigInteger;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.EntityTransaction;
+import javax.persistence.Query;
+import net.javacrumbs.shedlock.core.SchedulerLock;
+import org.apache.commons.collections4.CollectionUtils;
+import org.elasticsearch.common.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.jmx.export.annotation.ManagedOperation;
+import org.springframework.jmx.export.annotation.ManagedResource;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import uk.nhs.tis.sync.event.JobExecutionEvent;
+
+
+/**
+ * Job for updating funding status for posts. This class extends the
+ * PersonDateChangeCaptureSyncJobTemplate and provides functionality to synchronize post funding
```

Comment:
N.B. Adding a reminder comment to check that the comment is updated to match the class relationship.